File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        int low = 1, high = x;
        int mid = 0;
        while (low <= high){
            mid = low + (high -low)/2;
            if (mid <= x /mid){
                low = mid +1;
            }    
            else{
                high = mid -1;
            }
        }
        return high;
    }
}

File: 0069-sqrtx/0069-sqrtx.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        low = 0
        high = x
        while low <= high:
            if x < 2:
                return x

            mid = low + (high - low)//2

            if mid <= x/mid:
                low = mid +1

            else:
                high = mid -1 

        return high